Source/NSGridView.m

```diff
@@ -0,0 +1,1262 @@
+              [v setFrame: rect];
+              [self addSubview: v];
+            }
+          
+          current_x += [col trailingPadding] + [col width] + [[colWidths objectAtIndex: ci] floatValue] + _columnSpacing;
+          current_y -= [row bottomPadding] - [row height]; // - _rowSpacing; // add paddings after view...
+          
+          // inc
+          i++;
+        }
+      END_FOR_IN(_cells);
+    }
+}
+
+- (instancetype) initWithFrame: (NSRect)frameRect
+{
+  self = [super initWithFrame: frameRect];
+
+  if (self != nil)
+    {
+      _rows = [[NSMutableArray alloc] init];
+      _columns = [[NSMutableArray alloc] init];
+      _cells = [[NSMutableArray alloc] init];
+    }
+
+  [self _refreshCells];
+  
+  return self;
+}
+
+- (instancetype) initWithViews: (NSArray *)views // an array of arrays
+{
+  self = [self initWithFrame: NSZeroRect];
+
+  NSUInteger c = 0;
+  NSUInteger r = 0;
+
+  if (self != nil)
+    {
+      NSUInteger rc = [views count], cc = 0;
+
+      if (rc > 0)
+        {
+          cc = [[views objectAtIndex: 0] count];
+        }
+      
+      _cells = [[NSMutableArray alloc] initWithCapacity: rc * cc];
+      _rows = [[NSMutableArray alloc] initWithCapacity: rc];
+      _columns = [[NSMutableArray alloc] initWithCapacity: cc];
+
+      FOR_IN(NSArray*, row, views)
+        {
+          NSArray *columns = [row objectAtIndex: c];
+          FOR_IN(NSArray*, column, columns)
+            {
+              NSView *v = [column objectAtIndex: c];
+              NSGridCell *cell = [[NSGridCell alloc] init];
+
+              if (r == 0)
+                {
+                  NSGridRow *gr = [[NSGridRow alloc] init];
+                  [gr setGridView: self];
+                  [_rows addObject: gr];
+                  [cell _setOwningRow: gr];
+                }
+
+              if (c == 0)
+                {
+                  NSGridColumn *gc = [[NSGridColumn alloc] init];
+                  [gc setGridView: self];
+                  [_columns addObject: gc];
+                  [cell _setOwningColumn: gc];
+                }
+              
+              [cell setContentView: v];
+              [_cells addObject: cell];
+              c++;
+            }
+          END_FOR_IN(columns);
+          r++;
+        }
+      END_FOR_IN(rows);
+      [self _refreshCells];
+    }  
+  return self;
+}
+
++ (instancetype) gridViewWithNumberOfColumns: (NSInteger)columnCount rows: (NSInteger)rowCount
+{
+  NSUInteger r = 0;
+  NSUInteger c = 0;
+  NSMutableArray *rows = [[NSMutableArray alloc] initWithCapacity: rowCount];
+  
+  for (r = 0; r < rowCount; r++)
+    {
+      NSMutableArray *col = [[NSMutableArray alloc] initWithCapacity: columnCount];
+      for (c = 0; c < columnCount; c++)
+        {
+          NSGridCell *gc = [[NSGridCell alloc] init];
+          [col addObject: gc];
+          RELEASE(gc);
+        }
+      [rows addObject: col];
+    }
+
+  return [self gridViewWithViews: rows];
```

Comment:
My comment here seems to be lost, so I add it again.
The method gridViewWithViews: expects an array of arrays of views. What you create here is an array of arrays pf NSGridCells. This is something completely different and won't work.

Reply:
Ah, thank you for pointing that out.  I have changed this to generate an array of arrays of NSViews and then call gridViewWIthViews:.   Sorry if this seems to have been lost before.